### ISSUE TITLE
Read timeout of complete DC per minute is larger than 5000 during refreshing

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#2956]
*Installation details*
Scylla version (or git commit hash): scylla-server-1.7.5-0.20170918.760af56.el7.centos.x86_64
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

Check condition: Expect `read timeout of complete DC per minute` should be less than `5000`.

*1.7.x(Result: Bad)*

- https://cloudius-jenkins-test.s3.amazonaws.com/local_run/job-2017-11-12T00.11-6b90418.png
  scylla-server-1.7.1-0.20170607.dbbf99d.el7.centos.x86_64
  result: serious
![image](https://user-images.githubusercontent.com/309708/32707514-11014a6c-c7ea-11e7-9a3e-3eb153992f5f.png)

- https://cloudius-jenkins-test.s3.amazonaws.com/local_run/job-2017-11-12T02.18-5e1731d.png
  scylla-server-1.7.5-0.20170918.760af56.el7.centos.x86_64
  result: serious

![image](https://user-images.githubusercontent.com/309708/32707542-43c065fa-c7ea-11e7-98e6-138ac17e3979.png)

*scylla-server-2.0.0-0.20170929.e265c91.el7.centos.x86_64 (result: Good)*

- https://cloudius-jenkins-test.s3.amazonaws.com/local_run/job-2017-11-12T04.43-3901798.png
  result: very little read_timeout

- https://cloudius-jenkins-test.s3.amazonaws.com/local_run/job-2017-11-12T13.49-0ed2002.png
  result: little read_timeout

- https://cloudius-jenkins-test.s3.amazonaws.com/local_run/job-2017-11-12T17.57-e743648.png
  result: almost no read_timeout

*Workload info*
workloads (3 db nodes `n1-highmem-16`,  4 loaders `n1-standard-2`):
- prepare_stress_cmd: cassandra-stress write no-warmup cl=QUORUM n=10000000 -schema 'replication(factor=3)' -port jmx=6868 -mode cql3 native -rate threads=1000 -errors ignore -pop seq=1..10000000 -log interval=5
- stress_cmd: cassandra-stress mixed no-warmup cl=QUORUM duration=90m -schema 'replication(factor=3)' -port jmx=6868 -mode cql3 native -rate threads=`1000` -errors ignore -pop seq=1..10000000 -log interval=5   (`two c-s processes per loader`)

*Test steps:*
1. Prepare write workload
2. Start mixed workload
3. Get backup of SSTables, uncompress it to upload directory
4. Load newly placed SSTables to the system
5. Verify the SSTables are loaded by select
6. Flush all SSTables
7. Wait until the mixed workload finishes
8. Check read_timeout during the refresh

@slivne @roydahan

**Comment by slivne** [comment]
I think there are two issues ...

We are missing the cpu scheduler - that should potentially fix some
We may have an issue with backpressure as well.

Need to retest the scpu scheduler is merged

**Comment by amoskong** [comment]
@slivne Do you mean merge cpu scheduler to 2.x?
2.0 works well, the timeout only exists in 1.7.x, why do you set 2.2 milestone?

**Comment by tzach** [comment]
@slivne @amoskong is this solved in 2.2? @amoskong did you retest?

**Comment by slivne** [comment]
We have a similar issue with streaming when we add a lot of sstables at once - 

- For streaming we are working on a change that will create larger sstables
- For refresh - I do not think we have a solution when we are adding alot of ssatbles at once

**Comment by slivne** [comment]
Off strategy compaction should help in this case - we plan todo a compaction prior to adding the files - so this will mean fewer files will be loaded.

Needs to be rested once that is done https://github.com/scylladb/scylla/issues/5226

**Comment by bhalevy** [comment]
@amoskong this can be retested as off-strategy compaction for refresh is already done in 7351db7cab7bbf907172940d0bbf8b90afde90ba

**Comment by slivne** [comment]
@amoskong / @roydahan - in the 4.3 / 4.2 5000 tables test - did we have refresh - was it better ?

**Comment by amoskong** [comment]
> @amoskong / @roydahan - in the 4.3 / 4.2 5000 tables test - did we have refresh - was it better ?

The issue was not found in `5000-tables job`, but a Refresh nemesis in regular longevity. `5000` is the total timeout per minute.

BTW, The longevity-5000-tables uses ChaosMonkey, all nemesis will be executed randomly.
It has chance to run the refresh Nemesis.

**Comment by roydahan** [comment]
This is very old issue.
I think that there are newer issues that covers the behaviour of refresh.
With some we already dealt during 4.3.

I recommend to close this one.
